lib/hostip.c

```diff
@@ -529,6 +533,21 @@ enum resolve_t Curl_resolv(struct Curl_easy *data,
         return CURLRESOLV_ERROR;
     }
 
+#ifdef ENABLE_IPV6
+#if defined(USE_RESOLVE_ON_IPS) && TARGET_OS_OSX
```

Comment:
Should this perhaps rather be ...`&& defined(TARGET_OS_OSX)` ? It risks causing a warning otherwise when not defined.

Also: where is `TARGET_OS_OSX` defined? Is that from the macOS SDK?

Reply:
`TARGET_OS_OSX` is defined in `TargetConditionals.h` from the macOS SDK, yes.
I can include it explicitly in the file, currently it only seems to be included implicitly.

Reply:
A  few more minutes of research:
`TARGET_OS_OSX` is defined with value `0` for non-macOS targets, therefore a simple `#ifdef TARGET_OS_OSX` is not the best way forward.
`#if defined(TARGET_OS_OSX) && TARGET_OS_OSX == 1` on the other hand seems to do the job.